### PR TITLE
Ignore invalid command class version during interview

### LINKED ISSUE #35: Invalid command class version from device cause Exception
As I'm evaluating WZWave I'm getting some exceptions as I'm starting to add devices.

```
com.whizzosoftware.wzwave.ZWaveRuntimeException: MULTI_CHANNEL_END_POINT_GET is not available in command class version 1
	at com.whizzosoftware.wzwave.commandclass.MultiInstanceCommandClass.createMultiChannelEndPointGet(MultiInstanceCommandClass.java:275)
	at com.whizzosoftware.wzwave.commandclass.MultiInstanceCommandClass.queueStartupMessages(MultiInstanceCommandClass.java:135)
	at com.whizzosoftware.wzwave.node.ZWaveNode.setState(ZWaveNode.java:145)
	at com.whizzosoftware.wzwave.node.ZWaveNode.onApplicationCommand(ZWaveNode.java:184)
	at com.whizzosoftware.wzwave.controller.netty.NettyZWaveController.onApplicationCommand(NettyZWaveController.java:399)
	at com.whizzosoftware.wzwave.channel.ZWaveChannelInboundHandler.channelRead(ZWaveChannelInboundHandler.java:56)
	at com.whizzosoftware.wzwave.channel.ZWaveChannelInboundHandler.userEventTriggered(ZWaveChannelInboundHandler.java:94)
	at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:304)
	at io.netty.channel.AbstractChannelHandlerContext.fireUserEventTriggered(AbstractChannelHandlerContext.java:290)
	at com.whizzosoftware.wzwave.channel.TransactionInboundHandler$NettyZWaveChannelContext.process(TransactionInboundHandler.java:171)
	at com.whizzosoftware.wzwave.channel.TransactionInboundHandler.channelRead(TransactionInboundHandler.java:66)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:333)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:319)
	at com.whizzosoftware.wzwave.channel.FrameQueueHandler.channelRead(FrameQueueHandler.java:239)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:333)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:319)
	at com.whizzosoftware.wzwave.channel.ACKInboundHandler.channelRead(ACKInboundHandler.java:33)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:333)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:319)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:163)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:333)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:319)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:787)
	at io.netty.channel.oio.AbstractOioByteChannel.doRead(AbstractOioByteChannel.java:147)
	at io.netty.channel.oio.AbstractOioChannel$1.run(AbstractOioChannel.java:45)
	at io.netty.channel.ThreadPerChannelEventLoop.run(ThreadPerChannelEventLoop.java:53)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:116)
	at java.lang.Thread.run(Thread.java:744)
```

It turns out this is happening because the device in this case announce support for the MultiInstanceCommandClass. But when WZWave query the device for supported command class version, the response will indicate the command class is not supported by saying that supported version is 0x00.

```
14:15:35.279 [pool-1-thread-1] TRACE c.w.wzwave.codec.ZWaveFrameDecoder - Done processing received data: [ZW_APPLICATION_COMMAND_HANDLER[0x05]: 0x86 0x14 0x60 0x00 ]
...and later...
14:15:35.363 [pool-1-thread-1] DEBUG c.w.w.c.VersionCommandClass - Setting command class COMMAND_CLASS_MULTI_INSTANCE to version 0x00
```

The specs are not very clear how the host should handle this scenario. But version 0 is never valid, and so I propose to ignore response and continue with default (version 1).

### PR DESCRIPTION
**Pull request by eperott**
Closes #35

### CODE REVIEW

**Comment by whizzosoftware** [comment]
Thanks for the change! Could you please add a test method to VersionCommandClassTests to confirm expected behavior?

File: src/main/java/com/whizzosoftware/wzwave/commandclass/VersionCommandClass.java [hunk 1]
```diff
@@ -61,20 +61,21 @@ public void onApplicationCommand(NodeContext context, byte[] ccb, int startIndex
         if (ccb[startIndex+1] == VERSION_REPORT) {
             int start = startIndex+2;
             library = String.format("%d", ccb[start]);
-            protocol = String.format("%d.%2d", ccb[start + 1], ccb[start + 2]);
-            application = String.format("%d.%2d", ccb[start + 3], ccb[start + 4]);
+            protocol = String.format("%d.%02d", ccb[start + 1], ccb[start + 2]);
```
Comment (eperott) [inline src/main/java/com/whizzosoftware/wzwave/commandclass/VersionCommandClass.java]: Programmers guide suggest padding with zeroes ('0'), so added this here. Let me know your thoughts on this.

File: src/main/java/com/whizzosoftware/wzwave/commandclass/VersionCommandClass.java [hunk 2]
```diff
@@ -61,20 +61,21 @@ public void onApplicationCommand(NodeContext context, byte[] ccb, int startIndex
...
+            application = String.format("%d.%02d", ccb[start + 3], ccb[start + 4]);
             logger.debug("Node {} uses library {}, protocol {} and application {}", ByteUtil.createString(context.getNodeId()), library, protocol, application);
-        } else if (ccb[1] == VERSION_COMMAND_CLASS_REPORT) {
+        } else if (ccb[startIndex+1] == VERSION_COMMAND_CLASS_REPORT) {
```
Comment (eperott) [inline src/main/java/com/whizzosoftware/wzwave/commandclass/VersionCommandClass.java]: Added startIndex offset here as well.

**Review by eperott — COMMENTED** [review]
Added test cases to cover callback behavior. I've added a couple of comments where I've changed behavior a bit. Please verify this is OK.

**Pull request merged**